src/components/Input/Input.tsx

```diff
@@ -0,0 +1,177 @@
+import React, { FC, useState } from "react";
+import styled, { css } from "styled-components";
+import { Color } from "../../theme/util";
+import { Row, Box, Spacer, Column } from "../../theme/custom-styled-components";
+import { ErrorIcon, CheckIcon } from "../Icon/Icons";
+import { InputProps, StyledLabelProps, StyledInputProps, LabelContainerProps } from "./InputTypes";
+
+const StyledLabel = styled.label<StyledLabelProps>`
+  font-family: sans-serif;
+  z-index: 1;
+  padding: 16px 5px 16px 0;
+  font-size: 17px;
+  color: ${Color.TextPrimary};
+
+  ${(p) =>
+    p.hasError &&
+    css`
+      color: ${Color.ErrorRed};
+    `}
+  /* TODO remove 'all', animate only 'fontsize' and 'top'*/
+    transition: all 0.3s ease;
+`;
+
+const LabelContainer = styled.div<LabelContainerProps>`
+  display: flex;
+  flex-direction: row;
+  pointer-events: none;
+  position: absolute;
+  align-items: center;
+  z-index: 6;
+  pointer-events: ${(p) => (p.disabled ? "none" : "auto")};
+  padding-left: 16px;
+
+  ${(props) =>
+    props.hasFocus &&
+    css`
+      top: ${(p: { size: number }) => `-${p.size / 4 - 3}px`};
```

Comment:
What about use using `1` instead of calculating `4-3` if the value that we are dividing by is always the same.

Reply:
I believe it is reading is as 4 `(p.size / 4) - 3`, (that's intended, and the parentheses were automatically excluded)